src/core/ready.js

```diff
@@ -5,12 +5,13 @@ define( [
 ], function( jQuery, document ) {
 
 // The deferred used on DOM ready
-var readyList;
+var readyList = jQuery.Deferred(),
+	readyPromise = readyList.promise();
 
 jQuery.fn.ready = function( fn ) {
 
 	// Add the callback
-	jQuery.ready.promise().done( fn );
+	readyPromise.done( fn );
```

Comment:
This `.done` is the source of #1823... if we're willing to give up a synchronicity guarantee, then it could become `.then`, the [synchronous invocations](https://github.com/jquery/jquery/pull/2891/files#diff-80299fe5124dec96e0bf1341cfcceb49R64) could get wrapped in `setTimeout` (removing `try...finally` in the process), and those [error resiliency tests](https://github.com/timmywil/jquery/commit/08d6e04d43b257284b9306fbb85a3e6423bac972#diff-11bce668df1353212ad70cea7ca6ceed) could be restored.

What do you think? Is anyone counting on `jQuery( invokedFirst ); invokedSecond()`?